### DISCUSSION TITLE
Planned discontinuation of imgui-go

### DISCUSSION THREAD

**Discussion by dertseha** [inkyblackness/imgui-go#192] [Announcements]
I have decided to officially discontinue **imgui-go**. With 2022-12-31, this repository will be set to read-only.
As an alternative, I recommend people switch over to https://github.com/AllenDang/cimgui-go instead.

The reasons for this move were accumulating, such as lack of time for maintenance or even lack of own use being the main drivers. With the creation of AllenDang/cimgui-go , there was finally a solution to what I considered myself to be a better way for the ever-changing **Dear ImGui** library.

When the currently pending pull-request about `FontGlyph` has been handled, there will be another release that also contains the sunsetting announcement in the readme. Any further extensions or fixes will be accepted until the end of the year. At this point I'll then do a final release where `imgui.CreateContext()` will be marked deprecated with a corresponding notice.

**Comment by kkevinchou** [comment]
Hello! Just wanted to stop by and say thank you for all the work you've put into the project. It's helped immensely with some of my personal projects and it's been a pleasure to use. Best of luck in the future!

**Comment by eliasdaler** [comment]
Thanks for all the hard work!

I'm ready to take on some maintenance of this library as I depend a lot on it. I'll work on cimgui-go too, but currently it's not as usable as imgui-go, unfortunately.

Are you aware of any ongoing forks of imgui-go?

If not, I'm thinking of making a new fork which will attempt to be a new upstream.

It would possibly be good to link to my fork in the README before this repo gets archived because otherwise it'll be hard for people to find the "currently maintained" version when they stumble upon this repo in the future.

**Reply by dertseha** [reply]
Thank you for the message!
There are several forks of the repository, look at the "insights" then "network" tab. Though I don't see any recent fork that kept up to date with newest changes.
If you intend to keep the thing going, it's also great.

**Reply by eliasdaler** [reply]
The problem with not picking a "successor" is that a "network" tab doesn't really show any meaningful detail.
So when people see an archived repo they might have contributed to, it's hard to find *which* repo to contribute to, because several forks might move along and add new commits.

As I didn't find the one who continued development, I'm willing to use my fork and accept PRs/answer issues once they arrive (I will add some new bindings later, just didn't have the time). If there was a chosen "successor" fork already, I would've contributed to it in the future, I didn't find any.

**Reply by dertseha** [reply]
I believe there's a misunderstanding; My reference to the network was to your question whether I knew there were any fork. The network is all I am aware of. I was not intending that visitors should use this as the sole lead for a successor.

As you went through them already now, I'm happy to refer to your repository in the readme. 
Which one shall I refer to? Your [GitHub linked-fork](https://github.com/eliasdaler/imgui-go), or are you following your initial idea of hard-copying a new one?

**Reply by eliasdaler** [reply]
Yes, this one is fine. 

After some digging, I've found that Github forks only gets removed if it gets DMCA'd. :)
I was thinking of making a hard fork only in terms of preservation, but it seems such worries are not needed.
